tests/helpers/fund-accs/run_full.go

```diff
@@ -138,18 +138,25 @@ func RunFullNode(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 
 	runenv.RecordMessage("full -> %d has this %s balance", initCtx.GroupSeq, bal.String())
 
-	nid, _ := hex.DecodeString("0c204d39600fddd3")
-	data := []byte("f1f20ca8007e910a3bf8b2e61da0f26bca07ef78717a6ea54165f5")
-	for i := 0; i < 10; i++ {
-		tx, err := nd.StateServ.SubmitPayForData(ctx, nid, data, 70000)
+	var nid namespace.ID
+	if runenv.StringParam("namespace-id") == "1" {
+		nid = common.DefaultNameId
+	} else {
+		nid = common.GetRandomNamespace()
+	}
+
+	data := common.GetRandomMessageBySize(runenv.IntParam("msg-size"))
+	for i := 0; i < runenv.IntParam("submit-times"); i++ {
+		err = common.SubmitData(ctx, runenv, nd, nid, data)
 		if err != nil {
 			return err
 		}
 
-		runenv.RecordMessage("code reponse is %d", tx.Code)
-		runenv.RecordMessage(tx.RawLog)
-		if tx.Code != 0 {
-			return fmt.Errorf("failed pfd")
+		if runenv.TestCase == "get-shares-by-namespace" {
+			err = common.CheckSharesByNamespace(ctx, nd, nid, data)
+			if err != nil {
+				return err
+			}
 		}
```

Comment:
this looks like the same function as https://github.com/celestiaorg/test-infra/pull/113/files#diff-f422ccf6c1986c5e7dcc11013483a122c85e0b44131e445f50a9fa8c17b84bf3R110

good make a helper

Reply:
looks like it is repeated several times. definitely worth making a helper to DRY it up

Reply:
I've cleaned up for this PR. Still, I see that we need to move to DI style as a lot of places we are just using similar code but need different `Options` to make the behaviour differ